### PR TITLE
Fix authPolicy remove

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

**What this PR does / why we need it**:

Fix the issue: a authorization deleted, the same namespace scoped authorization will be mis removed

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by supercharge-xsy** [comment]
The modification is fine, but seems that the test case needs to be modified accordingly.

File: pkg/auth/rbac.go [hunk 1]
```diff
@@ -160,14 +160,16 @@ func (r *Rbac) RemovePolicy(policyKey string) {
...
-		dstWorkload = r.workloadCache.GetWorkloadByAddr(cache.NetworkAddress{
-			Network: conn.dstNetwork,
-			Address: nets.ConvertIpByteToUint32(conn.dstIp),
-		})
+	dstWorkload := r.workloadCache.GetWorkloadByAddr(cache.NetworkAddress{
+		Network: conn.dstNetwork,
+		Address: nets.ConvertIpByteToUint32(conn.dstIp),
+	})
+	// If no workload found, deny
+	if dstWorkload == nil {
```
Comment (hzxuzhonghu) [inline pkg/auth/rbac.go:168]: @supercharge-xsy This is to align with ztunnel

**Pull request merged**